Client/game_sa/CBuildingSA.cpp

```diff
@@ -11,8 +11,56 @@
 
 #include "StdInc.h"
 #include "CBuildingSA.h"
+#include <game/CWorld.h>
+#include "CGameSA.h"
+
+extern CGameSA* pGame;
 
 CBuildingSA::CBuildingSA(CBuildingSAInterface* pInterface)
 {
     SetInterface(pInterface);
 }
+
+void CBuildingSA::SetLod(CBuilding* pLod)
+{
+    if (pLod)
+    {
+        if (m_pInterface->m_pLod)
+        {
+            SetLod(nullptr);
+        }
+
+        CBuildingSAInterface* pLodInterface = (dynamic_cast<CBuildingSA*>(pLod))->GetBuildingInterface();
```

Comment:
I would add` assert(pLodInterface)` there to make sure that this piece of code is always correct and for an early `nullptr` detection.